src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -429,24 +429,29 @@ internal static string DecodeStream(Stream stream, ref Encoding encoding)
             string content = StreamToString(stream, encoding);
             if (isDefaultEncoding)
             {
-                do
+                // No need to parse all content if it is large.
+                var count = Math.Min(2048, content.Length);
```

Comment:
You could easily have an embedded style sheet large this.   

The issue says the response header said it was text.  We should not be running this code on text.

Reply:
Since we can not implement full HTML parser we can only address common scenarios. From my understanding first tag always `<html>`, then `<head> and then common recommendation is to have `<meta>` tag on top place as best practice. The code follows this.
Also best practice is to have encoding tag in HTTP header so here we address an edge case in which the best practice is ignored.

Reply:
If we put a limit like this in, we should have an error or warning when it exceeded and possibly expose the limit value as a parameter so people can make it work.